plugin/src/test/java/io/jenkins/plugins/redis/RedisFingerprintStorageTest.java

```diff
@@ -0,0 +1,134 @@
+
+    private Jedis jedis;
+    private Map<String, String> savedSystemProperties = new HashMap<>();
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public GenericContainer redis = new GenericContainer<>("redis:6.0.4-alpine").withExposedPorts(6379);
+
+    @Before
+    public void setup() throws IOException {
+        preserveSystemProperty("redis.host");
+        preserveSystemProperty("redis.port");
+        preserveSystemProperty("FingerprintStorageEngine");
+
+        String address = redis.getHost();
+        Integer port = redis.getFirstMappedPort();
+
+        System.setProperty("redis.host", address);
+        System.setProperty("redis.port", String.valueOf(port));
+        System.setProperty("FingerprintStorageEngine", "io.jenkins.plugins.redis.RedisFingerprintStorage");
+
+        jedis = new Jedis(address, port);
+    }
+
+    @After
+    public void teardown() {
+        jedis.close();
+
+        for (Map.Entry<String, String> entry : savedSystemProperties.entrySet()) {
+            if (entry.getValue() != null) {
+                System.setProperty(entry.getKey(), entry.getValue());
+            } else {
+                System.clearProperty(entry.getKey());
+            }
+        }
+    }
+
+    private void preserveSystemProperty(String propertyName) {
+        savedSystemProperties.put(propertyName, System.getProperty(propertyName));
+        System.clearProperty(propertyName);
+    }
+
+    @Test
+    public void checkFingerprintStorageIsRedis() throws IOException {
+        Object fingerprintStorage = FingerprintStorage.get();
+        assertThat(fingerprintStorage, instanceOf(RedisFingerprintStorage.class));
+    }
+
+    @Test
+    public void roundTrip() throws IOException {
+        byte[] md5 = Fingerprint.toByteArray(Util.getDigestOf("roundTrip"));
+        Fingerprint fingerprintSaved = new Fingerprint(null, "foo.jar", md5);
+        Fingerprint fingerprintLoaded = Fingerprint.load(md5);
+        assertThat(fingerprintLoaded, is(not(nullValue())));
+        assertThat(fingerprintSaved.toString(), is(equalTo(fingerprintLoaded.toString())));
+    }
+
+    @Test
+    public void loadingNonExistentFingerprintShouldReturnNull() throws IOException{
+        byte[] md5 = Fingerprint.toByteArray(Util.getDigestOf("loadingNonExistentFingerprintShouldReturnNull"));
+        Fingerprint fingerprint = Fingerprint.load(md5);
+        assertThat(fingerprint, is(nullValue()));
+    }
+
+    @Test
+    public void shouldFailWhenStoredObjectIsInvalidFingerprint() throws IOException {
+        byte[] md5 = Fingerprint.toByteArray(Util.getDigestOf("shouldFailWhenStoredObjectIsInvalidFingerprint"));
+        String instanceId = Util.getDigestOf(new ByteArrayInputStream(InstanceIdentity.get().getPublic().getEncoded()));
+        jedis.set(instanceId+Util.toHexString(md5), "Invalid Data");
+        try {
+            Fingerprint.load(md5);
+        } catch (IOException ex) {
```

Comment:
You can simplify this by putting `expected = IOException.class` in @Test. You can probably test the exception string these days too via annotation